signingscript/src/signingscript/rcodesign.py

```diff
@@ -100,12 +101,33 @@ async def rcodesign_notary_wait(submission_id, creds_path):
         submission_id,
     ]
     log.info(f"Polling Apple Notary service for notarization status. Submission ID {submission_id}")
-    exitcode, _ = await _execute_command(command)
+    exitcode, logs = await _execute_command(command)
     if exitcode > 0:
         raise RCodesignError(f"Error polling notary service. Exit code {exitcode}")
+
+    await rcodesign_check_result(logs)
     return
 
 
+async def rcodesign_check_result(logs):
+    """Checks notarization results
+    rcodesign cli call can exit with 0 even though the notarization has failed
+    Args:
+        logs (list<str>): output from polling result
+    """
+    re_inprogress = re.compile("^poll state after.*InProgress$")
+    re_accepted = re.compile("^poll state after.*Accepted$")
+    re_invalid = re.compile("^poll state after.*Invalid$")
+    for line in logs:
+        if re_inprogress.search(line):
```

Comment:
I think you could skip the check for in-progress entirely, but that's only an advantage if the log is very long and performance is a concern...and I suppose it is nice to be explicit about the common case.

Reply:
It's not set to check the logs as they come in, only after the process is done.
So the log lines would look like this:
```
poll state after 50s: InProgress
poll state after 53s: InProgress
poll state after 57s: InProgress
poll state after 60s: InProgress
poll state after 64s: Accepted
.... <result logs>
```
So the `continue` on the first few lines, then `Accepted` on the last one, which returns the function.